iocage_lib/ioc_start.py

```diff
@@ -1500,7 +1502,9 @@ def find_bridge_mtu(self, bridge):
 
         memberif = self.get_bridge_members(bridge)
         if not memberif:
-            return '1500'
+            if self.unit_test:
+                return "1500"
```

Comment:
Can you please elaborate why we have this ? Wouldn't the default be `1500` and if no, don't you think this should be handled in the unit test rather then here ?

Reply:
I'll take a look at this further. The patch changes the default to be set by a variable, but that variable doesn't get set in unit_test mode. There were other places in the code where this stubbing out for test mode happened, so it seemed to be the way it was done. It didn't seem right to me, but seemed to be a convention. Anyway, leave it with me and I'll come up with something more elegant, because I didn't like that when I did it, so I'm glad there is a better way of handling unit_test mode.

In the mean time, it seems that Cirrus CI is failing on installing packages (ie. before our tests start).

Reply:
@sonicaj Ok, I've reproduced the error when I remove that "if self.unit_test: return "1500". It needs that because otherwise we get

> AttributeError: 'IOCStart' object has no attribute 'get'

in when running test mode (tests/unit_tests/1000_lib_start_test.py:55). The patch I'm looing to have merged makes the default MTU configurable, and that gets done with an IOCStart.get() call. So, I can see three options:

1. Leave it as is, with a hardcoded default that only services the unit test. This is done 30 lines above in the same function (dealing with DHCP being unavailable in unit test mode).
2. Remove the test. The test is hardcoded to test for "1500", and hardcoding the return value in #1 has very limited value in testing.
3. Remove the feature of having a default MTU other than 1500. The value of the default used to be hardcoded to 1500 (in fact, the whole thing was hardcoded to 1500 without this patch), so having a hardcoded default as long as you can configure per-interface is still useful.

I'd personally recommend solution #2. Remove what is a test which has limited value, particularly given the default can be changed. Are you ok with that being the resolution?

Reply:
Actually, please don't do #3. Having a default MTU setting is actually more useful than having per-interface. Chances are, in a given environment, it'll be the same for all interfaces, so you'll end up setting each and every interface to whatever the value you need is. Possible, but more painful.

Reply:
@sonicaj I've just committed an update which I'd like reviewed. I've provided a dummy IOCStart.get() for that test. So, the test remains, and the test does something useful. Could you please review again?

I'm just waiting for Cirrus-CI to finish executing. I needed to update it to an 11.3-STABLE snapshot in order to get it to execute. The version of the packages in the FreeBSD package repo are for 11.3, and don't install on 11.2 any more, it seems.

Reply:
Ok. Test is fixed so that it does test the functionality, and the CirrusCI tests pass. I'll resolve this thread.